### DISCUSSION TITLE
Reason: Message must be shorter than 262144 bytes.

### DISCUSSION THREAD

**Discussion by jktan0504** [co-go/sqs-extended-client-go#82] [Q&A] [answered]
Hi dear author, I am trying your solution. However, I still get the same error Message 
`Reason: Message must be shorter than 262144 bytes.`

Here is my sqs client configuration

```
var (
	client *sqsextendedclient.Client
)

func init() {

	cfg, err := config.LoadDefaultConfig(context.TODO())
	if err != nil {
		logging.Logger.Error("failed to get CFG: 28",
			zap.Error(err),
		)
	}

	// S3 Bucket
	s3Bucket := os.Getenv("STORAGE_S3_BUCKET")

	// Extended SQS Client
	// create a new sqsextendedclient with some options
	client, _ = sqsextendedclient.New(
		sqs.NewFromConfig(cfg),
		s3.NewFromConfig(cfg),
		// Get the bucket name
		sqsextendedclient.WithS3BucketName(s3Bucket),
		// set the threshold to 2gb
		sqsextendedclient.WithMessageSizeThreshold(2097152),
	)
}
```

Appreciated much if you can help me. Thanks expert.
![Screenshot 2024-02-16 at 3 31 40 PM](https://github.com/co-go/sqs-extended-client-go/assets/29939040/ae9098fe-3dd2-4243-bf9c-2287f4ee7a04)

**Comment by co-go** [comment, marked as answer]
Hey @jktan0504, thanks for trying out the library! In your initiation, it looks like you set the message threshold to 2 GB (`sqsextendedclient.WithMessageSizeThreshold(2097152)`). Unfortunately, AWS has a hard limit and will only accept _messages_ [up to 256 KiB](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/quotas-messages.html#) (look under the Message Size section). They accept _payloads_ up to 2 GB (so you can batch together multiple 256 KiB messages in a `SendMessageBatch` call), however it is currently impossible for any single message to exceed 256 KiB.

If you remove that line (configuring the message threshold to 2 GB), the default message size threshold is set to the maximum possible (256 KiB). That should result in sending your messages as expected.
